### PR TITLE
Updates seldon resources to match kubeflow release v1.4.0

### PR DESCRIPTION
**Pull request by DnPlas**
This PR introduces the following changes:
* examples/: updates serve-simple to v1
* src/templates: updates resources to seldon 1.6.0

### CODE REVIEW

**Comment by ca-scribner** [comment]
Should the integration tests be failing?  That might be related to charmed-kubernetes/actions-operator#25 and fixed by charmed-kubernetes/actions-operator#26.  If they're just failing on github, do they run ok locally?

**Comment by DnPlas** [comment]
> Should the integration tests be failing? That might be related to [charmed-kubernetes/actions-operator#25](https://github.com/charmed-kubernetes/actions-operator/issues/25) and fixed by [charmed-kubernetes/actions-operator#26](https://github.com/charmed-kubernetes/actions-operator/pull/26). 

It is slightly diff, in this case it timesout with msg `timed out waiting for the condition on deployments/hostpath-provisioner`, but it seems to be related. What I did to not care about that was to update the CI in #8 by initializing microk8s with `charmed-kubernetes/actions-operator@main`. #8 is not failing in that stage.

> If they're just failing on github, do they run ok locally?

@ca-scribner Short answer, yes tests in root/tests/ run ok locally.
Long ans: As it is right now, seldon-core-operator does not have your usual integration testing, just what seems to be [unit](https://github.com/canonical/seldon-core-operator/blob/master/tests/test_operator.py). They succeed in my local (famous last words), but looking at the logs, the issue is in the 'Twidle with Microk8s' stage rather than in the actual test stage, so tests didn't even run.

**Comment by DomFleischmann** [comment]
This is one of the cases where I prefer 1 PR with 2 commits for both the CI fixes and the upgrade, I want the PRs to pass instead of merging them with failing tests. I suggest we put #8  into this PR and make the tests pass

**Comment by DnPlas** [comment]
> This is one of the cases where I prefer 1 PR with 2 commits for both the CI fixes and the upgrade, I want the PRs to pass instead of merging them with failing tests. I suggest we put #8 into this PR and make the tests pass

@DomFleischmann I'd rather keep unrelated PRs separate, IMHO it is a better practice to keep a set of small and related changes in single PR, and have multiple PRs touching different parts of the repo, like the CI, the src code, READMEs, etc. In this particular case, we can wait for #8 to be merged an re-trigger the CI, that shouldn't be a concern.

**Comment by DomFleischmann** [comment]
> > This is one of the cases where I prefer 1 PR with 2 commits for both the CI fixes and the upgrade, I want the PRs to pass instead of merging them with failing tests. I suggest we put #8 into this PR and make the tests pass
> 
> @DomFleischmann I'd rather keep unrelated PRs separate, IMHO it is a better practice to keep a set of small and related changes in single PR, and have multiple PRs touching different parts of the repo, like the CI, the src code, READMEs, etc. In this particular case, we can wait for #8 to be merged an re-trigger the CI, that shouldn't be a concern.

Sounds good, glad you fixed it!

File: src/templates/validate.yaml [hunk 1]
```diff
@@ -2,20 +2,19 @@ apiVersion: admissionregistration.k8s.io/v1beta1
...
...WxqY3kxallUQ0NBU0l3CkRRWUpLb1pJaHZjTkFRRUJCUUFEZ2dFUEFEQ0NBUW9DZ2dFQkFKOXJnOGVrN2lXc3cxYWt1WHh0TWJxU1FKbnYKUFR4S1FKTGFseGZmWUZ0ZGxtMnN6ZWhMK3I2V0w0ZDdZM2R2d2lwZFc4U2o1TkMrQ0NBT0grK0RPT2FNVkF0SApuYzA1V3VkV1ZucGxIeDRtdnFWeEgrWWpudzRhSjFISS9XVW16dDVmYkp3aWx6dFJGYWV5NU81ZGJibTd5SkxmCmdhMDdkRFJHOExQbVR3NkFyR1oyMkt4djF3c202Rks1WlczcHpTL0VvSHZoNUtvME5lNTI0V28rem9JWWtuNlgKWmNuSlBaNFk5d01rR3FVKzBqRnk5TU9maFNJSTE2TlB0U2QraDAyS292VVQ5U3pxSVNrNUFmUkN5V0ZSeE9NYQpoUEtqZWRmV1gzOENzT2NETGdxWk1wbHJxdy9pQmZYY2tZVmpJakpEUk1mL3oxQWhpTnJMQm1UbXRIMENBd0VBCkFhTkNNRUF3RGdZRFZSMFBBUUgvQkFRREFnS2tNQjBHQTFVZEpRUVdNQlFHQ0NzR0FRVUZCd01CQmdnckJnRUYKQlFjREFqQVBCZ05WSFJNQkFmOEVCVEFEQVFIL01BMEdDU3FHU0liM0RRRUJDd1VBQTRJQkFRQlZveFNzcWtoVwpyRnNhRldSSTFsVEpKSlRLRkFHalkwSXM1QlBieHpNbGVtK3dpRzdKWVg1NUJDeUlMWkVoNlJxbWJaQlpuRng4CmlDSzNVT3lUdmZoQmN0RUpiUGd4WldtTnFCd04ra2lYWE5maWNjdmQwSEs3eTBJRnEvU2FqSklyMkJMbHBiMHEKWk5WRzYvSUlmNXJJWlpZTmdsVThPbmtRblZmVVJsandPL3NlTHVOTzAyNWZ5MmNzV2R1MFRwb2pNRDJhdjBEdwp2Wm9LMUtjSFFLZ2t5eDhZakRXQ1dpbFl2OXpHWDRzejRsaDV2QW01d2owRW5nVGhCclg4a2gvK3JNNFpiQmIwCmlsaSt1OTNpaXVTcHFjVU1UV3lWUU1lU1NBU2dobnZyMytJL3lnNE00OGRRUk1ubkI2dkJHejJ6K2JaUGhFa0QKMTdMaDg3K3ZCSERzCi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K'
```
Comment (knkski) [inline src/templates/validate.yaml:15]: We should probably revert all of these `caBundle` changes back to `Cg==`. I believe this `caBundle` value will get overwritten, but the `Cg==` value makes it much more obvious when something goes wrong. We'll probably just have to remember to manually do this every time we update these files.

File: src/templates/service.yaml [hunk 2]
```diff
@@ -1,10 +1,20 @@
 apiVersion: v1
 kind: Service
 metadata:
+  labels:
+    app: seldon
+    app.kubernetes.io/instance: 'seldon-core'
+    app.kubernetes.io/name: 'seldon-core-operator'
```
Comment (knkski) [inline src/templates/service.yaml:7]: We should either drop this annotation, or set it to `{{ self.model.app.name }}`

File: src/templates/service.yaml [hunk 3]
```diff
@@ -1,10 +1,20 @@
...
+  namespace: {{ namespace }}
 spec:
   ports:
   - port: {{ webhook_port }}
     targetPort: {{ webhook_port }}
   selector:
     juju-app: {{ service }}
+    app.kubernetes.io/instance: seldon1
+    app.kubernetes.io/name: seldon
+    app.kubernetes.io/version: v0.5
```
Comment (knkski) [inline src/templates/service.yaml:19]: Is this value supposed to match the `1.6.0` value above?

File: src/templates/configmap.yaml [hunk 4]
```diff
@@ -1,93 +1,16 @@
...
-    }
-  storageInitializer: |-
-    {
-        "image" : "gcr.io/kfserving/storage-initializer:v0.4.0",
-        "memoryRequest": "100Mi",
-        "memoryLimit": "1Gi",
-        "cpuRequest": "100m",
-        "cpuLimit": "1"
-    }
+  credentials: '{"gcs":{"gcsCredentialFileName":"gcloud-application-credentials.json"},"s3":{"s3AccessKeyIDName":"awsAccessKeyID","s3SecretAccessKeyName":"awsSecretAccessKey"}}'
```
Comment (knkski) [inline src/templates/configmap.yaml:3]: Can you run each of these json blobs through an autoformatter? That will prevent horizontal scroll, and also minimize the diff. https://beautifier.io/ is pretty handy if you don't have an autoformatter already.

File: examples/serve-simple-v1alpha2.yml [hunk 5]
```diff
@@ -1,27 +0,0 @@
-apiVersion: "machinelearning.seldon.io/v1alpha2"
```
Comment (knkski) [inline examples/serve-simple-v1alpha2.yml:1]: In the CRD, we're still claiming that we support `v1alpha2`. Might be worth keeping this around to ensure that's true.

**Comment by knkski** [comment]
TBH I'd also rather see #8 in here, as a separate commit. Similar to how we updated the istio operators to 1.11 in https://github.com/canonical/istio-operators/pull/31/commits

File: src/templates/validate.yaml (hunk 1 above)
Comment (DnPlas) [reply src/templates/validate.yaml:15]: +1, I wasn't sure if that was intentional in the last version of this file. Will change it.

File: src/templates/service.yaml (hunk 2 above)
Comment (DnPlas) [reply src/templates/service.yaml:7]: Agree, will use the latter.

File: src/templates/service.yaml (hunk 3 above)
Comment (DnPlas) [reply src/templates/service.yaml:19]: Seems like it is that way for both [Kubeflow](https://github.com/kubeflow/manifests/blob/master/contrib/seldon/seldon-core-operator/base/resources.yaml#L4830) and [Seldon](https://github.com/SeldonIO/seldon-core/blob/master/helm-charts/seldon-core-operator/templates/service_seldon-webhook-service.yaml) manifests.

File: src/templates/configmap.yaml (hunk 4 above)
Comment (DnPlas) [reply src/templates/configmap.yaml:3]: Absolutely, I asked you guys for a formatter, but then forgot to format it. Hehe, will do.

File: examples/serve-simple-v1alpha2.yml (hunk 5 above)
Comment (DnPlas) [reply examples/serve-simple-v1alpha2.yml:1]: I couldn't find what you are saying in the CRD, do you mind pointing me there? If we are still supporting `v1alpha2`, sure we can keep it for testing in the future.

Comment (knkski) [reply examples/serve-simple-v1alpha2.yml:1]: @DnPlas Here's where the CRD defines the default of `v1`, but also some older versions, including `v1alpha2`:

https://github.com/canonical/seldon-core-operator/blob/cf8f0e61/src/templates/crd.yaml#L4425-L4435

Comment (DnPlas) [reply examples/serve-simple-v1alpha2.yml]: Oh, sweet. I added it back. Thanks!

File: src/templates/configmap.yaml (hunk 4 above)
Comment (DnPlas) [reply src/templates/configmap.yaml]: Updated it.

File: src/templates/service.yaml (hunk 3 above)
Comment (DnPlas) [reply src/templates/service.yaml]: I've removed almost the whole block, except for `app.kubernetes.io/name: seldon`. Let me know if that is better.

File: src/templates/validate.yaml (hunk 1 above)
Comment (DnPlas) [reply src/templates/validate.yaml]: This is done.

File: src/templates/service.yaml (hunk 2 above)
Comment (DnPlas) [reply src/templates/service.yaml]: This is done.

File: src/templates/validate.yaml [hunk 6]
```diff
@@ -2,20 +2,19 @@ apiVersion: admissionregistration.k8s.io/v1beta1
...
+    app.kubernetes.io/instance: 'seldon-core'
+    app.kubernetes.io/name: 'seldon-core-operator'
+    app.kubernetes.io/version: '1.6.0'
+  name: seldon-validating-webhook-configuration-kubeflow
 webhooks:
 - clientConfig:
     caBundle: Cg==
     service:
       name: seldon-webhook-service
-      port: {{ webhook_port }}
```
Comment (knkski) [inline src/templates/validate.yaml:18]: I think we'll still need this line and the other similar ones in this file, as IIRC the default is `443`, and Seldon defaults to `4443`.

Comment (DnPlas) [reply src/templates/validate.yaml:18]: +1, updating now.

Comment (DnPlas) [reply src/templates/validate.yaml]: Done, resolving this

**Review by knkski — APPROVED** [review]
LGTM

**Pull request merged**